peepingtom/core/alchemists/base.py

```diff
@@ -0,0 +1,19 @@
+from abc import ABC, abstractmethod
+
+
+class Alchemist(ABC):
```

Comment:
This looks good, should we add the input block type and output block types somewhere into the object itself? possibly as abstract properties?
That would mean
- you can easily see what the inputs and outputs are without relying on the class name
- easy type checking when setting the inputs and outputs